src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -51,6 +54,7 @@ public RegionInfo(string region, RegionAutodetectionSource regionSource)
             {
```

Comment:
In general, it will be better to remove shouldClearStaticCache which is there for test only and add another extension method or use reflection to change values for testing.

Reply:
I will create a tracking item for this